projects/extension/src/background/ConnectionManager.ts

```diff
@@ -215,6 +218,17 @@ export class ConnectionManager
     this.#client = undefined
   }
 
+  runAlarm(): void {
+    this.#networks
+      .filter((n) => relayChains.has(n.name))
```

Comment:
Please never use names or anything that comes from the chain spec as a way to identify a chain. Names are completely untrusted input. Please use only the value that smoldot returns when calling `addChain` to know which chains to load the database from.

I think the entire  ̀ConnectionManager` needs a rework.

Reply:
I agree with this comment - but I would avoid reworking the whole `ConnectionManager` in this PR. 
If it is fine for you - we can use the name for identification at the moment - and fix the whole thing in a following PR. What do you think?

Reply:
I think that reworking the `ConnectionManager` is super top notch priority, and is way more important than this database thing

Reply:
> I think that reworking the `ConnectionManager` is super top notch priority, and is way more important than this database thing

#570